### PR TITLE
Removed some gross inefficiencies in the TsdCompoundUnit implementation.

### PR DESCRIPTION
**Pull request by vjkoskela**

### CODE REVIEW

File: src/test/java/com/arpnetworking/metrics/impl/TsdCompoundUnitTest.java [hunk 1]
```diff
@@ -515,14 +515,15 @@ public void testGetName() {
...
                 .addNumeratorUnit(Units.BYTE)
                 .addDenominatorUnit(new MyCompoundUnitA())
                 .addDenominatorUnit(Units.SECOND)
-                .build()
-                .getName();
+                .build();
+        name = unit.getName();
         Assert.assertNotNull(name);
         Assert.assertEquals("(byte*byte*kilobyte)/(millisecond*second*second)", name);
+        Assert.assertEquals(name, unit.getName());
```
Comment (BrandonArp) [inline src/test/java/com/arpnetworking/metrics/impl/TsdCompoundUnitTest.java]: I think you're looking for assertSame here.

Comment (vjkoskela) [inline src/test/java/com/arpnetworking/metrics/impl/TsdCompoundUnitTest.java]: Fixed.

**Comment by matthayter** [comment]
Can we get a description of the inefficiencies?

**Comment by vjkoskela** [comment]
@matthayter the two issues addressed with this pull request are:

1) Cache the compound unit name which is used to ensure a consistent order of units during reduction (e.g. lb-ft == ft-lb).
2) Reducing common units no longer creates a key intersection set and instead iterates the smaller of the two sets.

**Pull request merged**